examples/pennylane_run_isingmodel_PyTorch.py

```diff
@@ -0,0 +1,215 @@
+        print("Energy after step {:5d}: {: .7f} | Angles: {}".format(i+1, costn, [p1n.detach().numpy(), p2n.detach().numpy()]),"\n")
+        
+
+###############################################################################
+#
+# .. note::
+#     When using the *PyTorch* optimizer, keep in mind that:
+#
+#     1. ``loss.backward()`` computes the gradient of the cost function with respect to all parameters with ``requires_grad=True``. 
+#     2. ``opt.step()`` performs the parameter update based on this *current* gradient and the learning rate. 
+#     3. ``opt.zero_grad()`` sets all the gradients back to zero. It's important to call this before ``loss.backward()`` to avoid the accumulation of gradients from multiple passes.
+#
+#     Hence, its standard practice to define the ``closure()`` function that clears up the old gradient, 
+#     evaluates the new gradient and passes it onto the optimizer in each step. 
+#
+# The minimum energy is -2 for the spin configuration :math:`[s_1, s_2, s_3] = [1, 1, -1]`
+# which corresponds to
+# :math:`(\phi, \theta, \omega) = (0, 0, 0)` for the second spin and :math:`(\phi, \theta, \omega) = (0, \pi, 0)` for 
+# the third spin. Note that gradient descent optimization might not find this global minimum due to the non-convex cost function, as is shown in the next section.
+
+p1_final, p2_final = opt.param_groups[0]["params"]
+print("Optimized angles:", p1_final, p2_final)
+print("Final cost after optimization:", cost(p1_final, p2_final))
+
+###############################################################################
+
+import matplotlib
+import matplotlib.pyplot as plt
+
+fig = plt.figure(figsize=(6, 4))
+plt.plot(x, cost_pt, label = 'global minimum')
+plt.xlabel("Optimization steps")
+plt.ylabel("Cost / Energy")
+plt.legend()
+plt.show()
+
+###############################################################################
+# Local minimum
+# ^^^^^^^^^^^^^
+# If the spins are initialized close to the local minimum of zero energy, the optimizer is
+# likely to get stuck here and never find the global minimum at -2. 
+
+torch.manual_seed(9)
+p3 = Variable((np.pi*torch.rand(3, dtype = torch.float64)), requires_grad = True)
+p4 = Variable((np.pi*torch.rand(3, dtype = torch.float64)), requires_grad = True)
+
+var_init_loc = [p3, p4]
+cost_init_loc = cost(p3, p4)
+
+print("Corresponding cost before optimization:", cost_init_loc)
+
+
+###############################################################################
+
+opt = torch.optim.SGD(var_init_loc, lr = 0.1)
+
+def closure():
+    opt.zero_grad()
+    loss = cost(p3, p4)
+    loss.backward()
+    return loss
+
+var_pt_loc = [var_init_loc]
+cost_pt_loc = [cost_init_loc]
+
+for j in range(100):
+    opt.step(closure)
+    if (j + 1) % 5 == 0:
+        p3n, p4n = opt.param_groups[0]['params']
+        costn = cost(p3n, p4n)
+        var_pt_loc.append([p3n, p4n])
+        cost_pt_loc.append(costn)
+
+        # for clarity, the anlges are printed as numpy arrays
```

Comment:
```suggestion
        # for clarity, the angles are printed as numpy arrays
```

Reply:
oh! good find! I read this document twenty times but did not catch this! :D